core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/impl/api/parse.kt

```diff
@@ -408,18 +542,44 @@ internal fun <T> DataColumn<String?>.parse(parser: StringParser<T>, options: Par
     return DataColumn.createValueColumn(name(), parsedValues, parser.type.withNullability(hasNulls)) as DataColumn<T?>
 }
 
-internal fun <T> DataFrame<T>.parseImpl(options: ParserOptions?, columns: ColumnsSelector<T, Any?>) =
-    convert(columns).to {
-        when {
-            it.isFrameColumn() -> it.cast<AnyFrame?>().parse(options)
-
-            it.isColumnGroup() ->
-                it.parse(options) { all() }
-                    .asColumnGroup(it.name())
-                    .asDataColumn()
-
-            it.typeClass == String::class -> it.cast<String?>().tryParse(options)
+internal fun <T> DataFrame<T>.parseImpl(options: ParserOptions?, columns: ColumnsSelector<T, Any?>): DataFrame<T> =
+    runBlocking { parseParallel(options, columns) }
+
+private suspend fun <T> DataFrame<T>.parseParallel(
+    options: ParserOptions?,
+    columns: ColumnsSelector<T, Any?>,
+): DataFrame<T> =
+    coroutineScope {
+        val convertedCols = getColumnsWithPaths(columns).map { col ->
+            async {
+                when {
+                    // when a frame column is requested to be parsed,
+                    // parse each value/frame column at any depth inside each DataFrame in the frame column
+                    col.isFrameColumn() ->
+                        col.values.map {
+                            async {
+                                it.parseParallel(options) {
+                                    colsAtAnyDepth { !it.isColumnGroup() }
+                                }
+                            }
+                        }.awaitAll()
+                            .toColumn(col.name)
+
+                    // when a column group is requested to be parsed,
+                    // parse each column in the group
+                    col.isColumnGroup() ->
+                        col.parseParallel(options) { all() }
+                            .asColumnGroup(col.name())
+                            .asDataColumn()
+
+                    // Base case, parse the column if it's a `String?` column
+                    col.isSubtypeOf<String?>() ->
+                        col.cast<String?>().tryParse(options)
```

Comment:
Are we throwing for now here some exceptions? in tryParse?

Reply:
No, `DataColumn.tryParse` means it tries to parse it, but if it fails, it will keep it as `String`. This is in contrast with `DataColumn.parse` which does throw an exception if it couldn't be parsed.

I'll add some quick kdoc, because it looks confusing indeed.